### PR TITLE
Compiler Documentation

### PR DESCRIPTION
**Pull request by isc30**
[Recommended link to check the diff](https://github.com/DerivcoIpswich/dsharp/pull/125/files?short_path=1e290ac#diff-1e290ac8433d555bce009b162cb869d0)

Add documentation about:
- project structure
- compiler implementation details
- testing
- debugging
- useful links

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -8,11 +8,105 @@ Specifically, DSharp lets you leverage the experience of C# (intellisense, build
 
 You can even use DSharp to work against other existing frameworks and APIs such as jQuery, jQuery plugins and Knockout, and can be extended to work against other existing script.
 
-DSharp requires .NET 4.0 and Visual Studio 2012.
+## Compatibility ##
+
+The current solution `DSharp.sln` is only compatible with .NET 4.6.1 and Visual Studio 2017 for Windows.
```
Comment (fred-perkins) [inline Readme.md]: This is going to change with some work that Kev is doing to add 4.0.

**Comment by isc30** [comment]
mono cecil

File: Readme.md (hunk 1 above)
Comment (isc30) [reply Readme.md]: done

File: Readme.md [hunk 2]
```diff
@@ -8,11 +8,106 @@ Specifically, DSharp lets you leverage the experience of C# (intellisense, build
...
+    ```
+    > update the paths to match the ones in your pc
+3. F5 in debug mode
+
+## Useful Links ##
+
+- ### mscorlib
+  Being compliant with latest implementations of `mscorlib` is extremely important for cross-compiling. These can be checked in the following links:
+
+  - [Source Dot Net](https://source.dot.net): Source reference with nice search and navigation
```
Comment (fred-perkins) [inline Readme.md]: Can we add Reference source here as well

File: Readme.md [hunk 3]
```diff
@@ -8,11 +8,106 @@ Specifically, DSharp lets you leverage the experience of C# (intellisense, build
...
+  - [GitHub Project](https://github.com/dotnet/coreclr/tree/master/src/System.Private.CoreLib): Latests previews for NET Core
+  - [Net Standard GitHub](https://github.com/dotnet/standard)
 
 ### Credits ###
 
-DSharp is a fork of the excellent Script#  project.
```
Comment (fred-perkins) [inline Readme.md]: Hah

File: Readme.md [hunk 4]
```diff
@@ -8,11 +8,106 @@ Specifically, DSharp lets you leverage the experience of C# (intellisense, build
...
+### Debugging the Shell ###
+
+Debugging the Shell requires a bit of configuration: setting up the arguments for the invokation.
+
+In order to debug the shell you need to:
+
+1. Rebuild All
+2. In the `Core/DSharp.Shell` project, `Right Click` -> `Properties` -> `Debug` -> `Application arguments` and set the following:
+
+    ```
```
Comment (fred-perkins) [inline Readme.md]: Formatting, also can you mark the code block with the syntax `console`

Comment (isc30) [reply Readme.md]: formatting is fine, since I want it to be indented. Adding the `console`

File: Readme.md (hunk 2 above)
Comment (isc30) [reply Readme.md]: what is the difference exactly?

Comment (fred-perkins) [reply Readme.md]: Not quite, Reference source is the source of .net 4.8.1. And source.net is the source of the current DotNetCore -> 2.2.x?

**Pull request merged**